pr_parser.py

```diff
@@ -0,0 +1,301 @@
+        return parsed_args
+
+    def assign_args(self, args):
+        """
+        Assign args to member variables.
+        :param args: Parsed args from the user
+        :return:
+        """
+        if args.repo:
+            self.__repo = args.repo
+        else:
+            print "\nMust specify repository url\n"
+            sys.exit(1)
+        
+        if args.target_branch:
+            self.__target_branch = args.target_branch
+        else:
+            print "\nMust specify a branch name\n"
+            sys.exit(1)
+        
+        if args.actual_commit:
+            self.__actual_commit = args.actual_commit
+        else:
+            print "\nMust specify an actual_commit\n"
+            sys.exit(1)
+
+        if args.merge_commit_sha:
+            self.__merge_commit_sha = args.merge_commit_sha
+        else:
+            print "\nMust specify a commit-id\n"
+            sys.exit(1)
+
+        if args.pull_id:
+            self.__pull_id = args.pull_id
+        else:
+            print "\nMust specify a pull-id\n"
+            sys.exit(1)
+        
+        
+    def parse_pr(self, base_repo, base_pull_id):
+        """
+        get related prs according to the base pr
+        :param repo: string, repo name associated with the pr to be parsed
+        :param pull_id: number in string, pull request id of the pr to be parsed
+        :return related_prs: list of tuple of string: [(repo, sha, pull_id, commit),...] 
+        pr list which is the associated with base pr
+        """
+
+        #init github  and get related pr object
+        HOME = os.environ['HOME']
+        with open('{0}/.ghtoken'.format(HOME), 'r') as file:
+            TOKEN = file.read().strip('\n')
+        gh = Github(TOKEN)
+        pr = gh.get_repo(base_repo).get_pull(long(base_pull_id))
+        
+        #get all comments and description in the pr
+        pr_texts = []
+        pr_texts.append(pr.body)
+        for pr_comment in pr.get_issue_comments():
+            pr_texts.append(pr_comment.body)
+        
+        #parse pr
+        related_prs = []
+        for pr_text in pr_texts:
+            pr_words = pr_text.lower().replace(':', '').split()
+
+            #find keyword "jenkins"
+            if 'jenkins' not in pr_words:
+                continue
+            position = pr_words.index('jenkins')
+
+            #analyse dependency relationship, "depend" or "ignore" 
+            if ('ignore' not in pr_words[position+1]) and ('depend' not in pr_words[position+1]):
+                continue
+            
+            #find "ignore"
+            if 'ignore' in pr_words[position+1]:
+                related_prs = None
+                print "INFO: \"Jenkins: ignore\" in repo: {0} pull_id: {1}".format(base_repo, base_pull_id)
+                break
+
+            #find "depend"
+            for i in range(position+2, len(pr_words)):
+                if 'https//github.com' not in pr_words[i]:
+                    break
```

Comment:
How about if the dependent PRs are separated by comma? As my understanding, your code will regard it as invalid input and fail testing, but I think this is a much common cases and we'd better support it.
For example:
```
Jenkins depends: xxxx, yyyyy, zzzzz
```